my-webxr-app/src/repository/DbRepository.tsx

```diff
@@ -0,0 +1,92 @@
+import {Repository} from "./Repository.tsx";
+import {DataPoint} from "./DataPoint.tsx";
+import Dexie from "dexie";
+import * as assert from "assert";
+
+export class Column {
+    id?: number;
+    name: string;
+    values: Array<number | string | null>;
+
+    constructor(name: string, values: Array<number | string | null>) {
+        this.name = name;
+        this.values = values;
+    }
+}
+
+export class DbRepository extends Dexie implements Repository {
+    // Declare implicit table properties.
+    // (just to inform Typescript. Instantiated by Dexie in stores() method)
+    private columns!: Dexie.Table<Column, number >; // number = type of the primkey
+    //...other tables goes here...
+
+    constructor(dbName: string) {
+        // create a db instance
+        super(dbName);
+
+        if (this.columns) {
+            this.columns.clear();
+        }
+
+        this.version(1).stores({
+            // Declare tables, IDs and indexes
+            // NOTE: Don’t declare all columns like in SQL.
+            // You only declare properties you want to index, that is properties you want to use in a where(…) query.
+            columns: '++id',
+        });
+    }
+
+    addColumn(column : Column) {
+        this.columns.add(column)
+            .catch(e => console.error("Error: " + e));
+    }
+
+    addColumnBulk(columns : Array<Column>) {
+        this.columns.bulkAdd(columns)
+            .catch(e => console.error("Error: " + e));
+    }
+
+    async getPoints(qualifyingPointOnly : boolean): Promise<Array<DataPoint>> {
```

Comment:
Add some documentation on what `qualifyingPointOnly` means, or refactor it into something like `filterIncompleteRows`

Reply:
done!